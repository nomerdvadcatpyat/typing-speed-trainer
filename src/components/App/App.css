* {
    box-sizing: border-box;
}

html, body, #root, .App {
    min-height: 100vh;
    padding: 0;
    margin: 0;
    font-family: 'Roboto', sans-serif;
}


.App {
    display: grid;
    grid-template:  'header' 10vh
                    'content' 1fr
                    / 1fr;
}