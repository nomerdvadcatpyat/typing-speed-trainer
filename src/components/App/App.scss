@import "./src/utils/scss-variables/variables";
@import "~bootstrap/scss/bootstrap";

* {
    box-sizing: border-box;
}

html, body, #root, .App {
    min-height: 100vh;
    padding: 0;
    margin: 0;
    font-family: 'Roboto', sans-serif;
    color: $dark;
}

.App {
    .main-spinner {
        position: absolute;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
        border-color: $dark;
    }
}