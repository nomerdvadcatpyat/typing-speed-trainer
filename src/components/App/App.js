import './App.css'
import { Header } from "../Header/Header";
import {Route} from 'react-router-dom';
import TrainingPageContainer from "../Pages/TrainingPage/TrainingPageContainer";

function App() {
  return (
    <div className="App">
      <Header />
      <Route path="/training">
        <TrainingPageContainer />
      </Route>

    </div>
  );
}

export default App;
