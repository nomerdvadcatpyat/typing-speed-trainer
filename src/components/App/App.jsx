import React from 'react';
import './App.scss'
import { Route, Switch } from 'react-router-dom';
import {AuthPage} from "../Pages/AuthPage/AuthPage";
import HeaderContainer from "../Header/HeaderContainer";
import {Ring} from "react-spinners-css";
import SearchRoomPageContainer from "../Pages/SearchRoomPage/SearchRoomPageContainer";
import CreateRoomPageContainer from "../Pages/CreateRoomPage/CreateRoomPageContainer";
import GamePageContainer from "../Pages/RoomPages/GamePage/GamePageContainer";
import WaitingRoomPageContainer from "../Pages/RoomPages/WaitingRoomPage/WaitingRoomPageContainer";
import {RoomPages} from "../Pages/RoomPages/RoomPages";
import RoomPagesContainer from "../Pages/RoomPages/RoomPagesContainer";
import {RatingPageContainer} from "../Pages/RatingPage/RatingPageContainer";

export function App({ isLoading, roomId }) {
  return (
      <div className="App">
        { isLoading ?
          <Ring className="main-spinner" /> :
          <>
          <HeaderContainer />
            <Switch>
              <Route path="/auth" component={AuthPage} />
              <Switch>
                  <Route path="/searchRoom" component={SearchRoomPageContainer} />
                  <Route path="/createRoom" component={CreateRoomPageContainer} />
                  <Route path="/room" component={RoomPagesContainer} />
                  <Route path="/rating" component={RatingPageContainer} />
              </Switch>
            </Switch>
          </>
        }
      </div>
  );
}

