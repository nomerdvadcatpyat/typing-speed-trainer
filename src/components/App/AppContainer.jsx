import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import {App} from "./App.jsx";
import {bindActionCreators} from "redux";
import {getIsAuth} from "../../store/selectors/userSelectors";
import {getIsLoading} from "../../store/selectors/appSelectors";
import {setUserActionCreator} from "../../store/actionCreators/userActionCreators";
import {initSocket, setLoadedState, setLoadingState} from "../../store/actionCreators/appActionCreators";
import {auth} from "../../utils/api/authApi";
import PropTypes from "prop-types";
import {getRoomId} from "../../store/selectors/gameSelectors";


const AppContainer = (props) => {
		useEffect(() => {
			props.setLoadingState();
			auth()
				.then(json => {
					if(json.ok)
						props.setUser(json.user)
				})
				.catch(console.log)
				.finally(props.setLoadedState)
		}, []);

		return (
			<App {...props} />
		);
}


const mapStateToProps = (state) => {
	return {
		isAuth: getIsAuth(state),
		isLoading: getIsLoading(state),
		roomId: getRoomId(state)
	};
}

const mapDispatchToProps = (dispatch) => {
	return {
		setLoadingState: bindActionCreators(setLoadingState, dispatch),
		setUser: bindActionCreators(setUserActionCreator, dispatch),
		setLoadedState: bindActionCreators(setLoadedState, dispatch),
		initSocket: bindActionCreators(initSocket, dispatch)
	}
}


export default connect(
	mapStateToProps,
	mapDispatchToProps
)(AppContainer);


AppContainer.propTypes = {
	isAuth: PropTypes.bool,
	isLoading: PropTypes.bool,
	setLoadingState: PropTypes.func,
	setUser: PropTypes.func,
	setLoadedState: PropTypes.func
}