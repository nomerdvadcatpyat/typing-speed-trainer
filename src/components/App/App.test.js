import React from 'react';
import {App} from "./App";
import { shallow, mount, render } from 'enzyme';

describe("<App />", () => {

	it("Should be render with loading", () => {
		const app = shallow(<App isLoading={true} isAuth={false}/>);
		expect(app).toMatchSnapshot();
	});

	it("App should be render with auth", () => {
		const app = shallow(<App isLoading={false} isAuth={true} />);
		expect(app).toMatchSnapshot();
	});

});