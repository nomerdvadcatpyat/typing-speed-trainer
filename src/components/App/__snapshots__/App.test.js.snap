// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App should be render with auth 1`] = `
<div
  className="App"
>
  <Connect(HeaderContainer) />
  <Switch>
    <Route
      component={[Function]}
      path="/training"
    />
  </Switch>
</div>
`;

exports[`<App /> Should be render with loading 1`] = `
<div
  className="App"
>
  <Ring
    className="main-spinner"
    color="#7f58af"
    size={80}
    style={Object {}}
  />
</div>
`;
